x/concentrated-liquidity/swaps_tick_cross_test.go

```diff
@@ -0,0 +1,741 @@
+	nextTick, err := types.TickIndexFromBytes(iter.Key())
+	s.Require().NoError(err)
+	s.Require().NoError(iter.Close())
+
+	s.Require().Equal(expectedTick, nextTick)
+}
+
+// assertPositionInRange a helper to assert that a position with the given lowerTick and upperTick is in range.
+func (s *SwapTickCrossTestSuite) asserPositionInRange(poolId uint64, lowerTick int64, upperTick int64) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	isInRange := pool.IsCurrentTickInRange(lowerTick, upperTick)
+	s.Require().True(isInRange, "currentTick: %d, lowerTick %d, upperTick: %d", pool.GetCurrentTick(), lowerTick, upperTick)
+}
+
+// assertPositionOutOfRange a helper to assert that a position with the given lowerTick and upperTick is out of range.
+func (s *SwapTickCrossTestSuite) assertPositionOutOfRange(poolId uint64, lowerTick int64, upperTick int64) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	isInRange := pool.IsCurrentTickInRange(lowerTick, upperTick)
+	s.Require().False(isInRange, "currentTick: %d, lowerTick %d, upperTick: %d", pool.GetCurrentTick(), lowerTick, upperTick)
+}
+
+// assertPositionRangeConditional a helper to assert that a position with the given lowerTick and upperTick is in or out of range
+// depending on the isOutOfRangeExpected flag.
+func (s *SwapTickCrossTestSuite) assertPositionRangeConditional(poolId uint64, isOutOfRangeExpected bool, lowerTick int64, upperTick int64) {
+	if isOutOfRangeExpected {
+		s.assertPositionOutOfRange(poolId, lowerTick, upperTick)
+	} else {
+		s.asserPositionInRange(poolId, lowerTick, upperTick)
+	}
+}
+
+// swapZeroForOneLeft swaps amount in the left (zfo) direction of the swap.
+// Asserts that no error is returned.
+func (s *SwapTickCrossTestSuite) swapZeroForOneLeft(poolId uint64, amount sdk.Coin) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	s.FundAcc(s.TestAccs[0], sdk.NewCoins(amount))
+	_, err = s.App.ConcentratedLiquidityKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool, amount, pool.GetToken1(), sdk.ZeroInt(), sdk.ZeroDec())
+	s.Require().NoError(err)
+}
+
+// swapOneForZeroRight swaps amount in the right (ofz) direction of the swap.
+// Asserts that no error is returned.
+func (s *SwapTickCrossTestSuite) swapOneForZeroRight(poolId uint64, amount sdk.Coin) {
+	pool, err := s.App.ConcentratedLiquidityKeeper.GetPoolById(s.Ctx, poolId)
+	s.Require().NoError(err)
+
+	s.FundAcc(s.TestAccs[0], sdk.NewCoins(amount))
+	_, err = s.App.ConcentratedLiquidityKeeper.SwapExactAmountIn(s.Ctx, s.TestAccs[0], pool, amount, pool.GetToken0(), sdk.ZeroInt(), sdk.ZeroDec())
+	s.Require().NoError(err)
+}
+
+// asserPoolLiquidityEquals a helper to assert that the liquidity of a pool is equal to the expected value.
+func (s *SwapTickCrossTestSuite) asserPoolLiquidityEquals(poolId uint64, expectedLiquidity sdk.Dec) {
```

Comment:
nit: the following assertions start with `asser` – is this intentional?
- `asserPositionInRange`
- `asserPoolLiquidityEquals`
- `asserPoolTickEquals`

Reply:
Fixed